cmd/agent/daemon/app/app.go

```diff
@@ -256,6 +261,48 @@ func (a *App) Run(ctx context.Context) error {
 		return fmt.Errorf("proc handler: %w", err)
 	}
 
+	if cfg.EBPFMetrics.ProgramMetricsEnabled {
+		isBPFEnabled, err := ebpftracer.BPFStatsEnabled()
+		if err != nil {
+			log.Errorf("cannot check if bpf stats are enabled: %v", err)
+			// I know that gotos are normally discouraged, but this is way more readable than
+			// having IFs nested three layers deep.
+			goto bpfDone
+		}
+
+		if isBPFEnabled {
+			goto bpfDone
+		}
+
+		err = ebpftracer.SetBPFStatsEnabled(true)
+		if err != nil {
+			log.Errorf("error while trying to enable bpf stats: %v", err)
+
+			// We disable program metrics in case of an error, as it will only cause error log messages
+			// everytime the metrics exporter runs.
+			cfg.EBPFMetrics.ProgramMetricsEnabled = false
+
+			goto bpfDone
+		}
+
+		// We need to disable bpf stats again, as the overhead is somewhat expensive.
+		defer func() {
```

Comment:
I am not 100% sure if I like this. If one runs the agent with this enabled, but then the agent gets `SIGKILL`ed, we would still have bpf stats enabled, meaning there would be a performance impact of the underlying host.